src/linagora.esn.calendar/app/planning/calendar-planning.controller.js

```diff
@@ -53,7 +53,10 @@ const _ = require('lodash');
 
             eventSourcesMap[calId] = buildEventSourceForCalendar(calendar);
             calendarPromise.then(function(cal) {
-              cal.fullCalendar('addEventSource', eventSourcesMap[calId]);
+              calendarVisibilityService.isHidden(calendar).then(isHidden => {
+                if (!isHidden) cal.fullCalendar('addEventSource', eventSourcesMap[calId]);
+              });
+              cal.fullCalendar('removeEventSource', eventSourcesMap[calId]);
```

Comment:
I suggest calling the `removeEventSource` first ( even tho it's going to be called first in your current code cause javascript won't wait for the `calendarVisibilityService`  call ). for me it looks more clear if we reflect it in the code 

```suggestion
              cal.fullCalendar('removeEventSource', eventSourcesMap[calId]); // remove it first
              calendarVisibilityService.isHidden(calendar).then(isHidden => {
                // if it's not hidden add it again to redraw
                if (!isHidden) cal.fullCalendar('addEventSource', eventSourcesMap[calId]);
              });

```